### ISSUE TITLE
Comment author avatar url is not correct when executing as a public user

### ISSUE THREAD

**Issue by umutesen** [wp-graphql/wp-graphql#2421]
I'm trying to load comments of a post along with comment author avatars. Avatar URL appears  to be incorrect when executing the query as a public user inside the GraphiQL IDE:

Executing as the logged in user returns:
```
"avatar": {
  "url": "http://2.gravatar.com/avatar/e8a00208a451c66eb52c105b9daa248a?s=96&d=mm&r=g"
}
```

Executing as a public user returns:
```
"avatar": {
  "url": "http://1.gravatar.com/avatar/?s=96&d=mm&r=g"
}
```

here is my query:
```
query NewQuery {
  postBy(slug: "my-post-slug") {
    comments {
      edges {
          node {
            id
            author {
              node {
                   avatar {
                    url
                  }
                }
            }
          }
        }
    }
  }
}
```

**Comment by jasonbahl** [comment]
@umutesen can you provide a bit more information to reproduce? For example maybe specific information about the user data you have in place? 

Would be good if I could get in a spot where I could reproduce this.

**Comment by jasonbahl** [comment]
@umutesen does this only happen when querying for a comment author? or does it happen when querying for a user directly?

The more information you can provide to help me reproduce this, the better. Thanks! 🙏🏻

**Comment by umutesen** [comment]
Many thanks for your reply. Comments have been captured by default wordpress comment system. The users leave comments without creating accounts so I am unable to query users directly. 

I am currently migrating to a headless WP setup with Angular as the front end, using Apollo client. All calls to wpgraphql is made without authentication.

The default avatar under discussion settings is **Identicon (generated)**. This generates a random avatar where the comment author does not have a gravatar associated with their email address.

<img width="738" alt="Screenshot 2022-06-25 at 09 41 50" src="https://user-images.githubusercontent.com/4234033/175765699-7e750914-f42b-4d17-8eca-3859df52bc9c.png">

To reproduce the error inside WP admin IDE, I query comments along with author avatars in a postBy query, executing as the logged in user.
"http://2.gravatar.com/avatar/e8a00208a451c66eb52c105b9daa248a?s=96&d=identicon&r=g" is a valid url.

<img width="1035" alt="Screenshot 2022-06-25 at 09 43 47" src="https://user-images.githubusercontent.com/4234033/175765771-a0bbcbac-1a74-4f85-86d7-6c8a1c3bd4ce.png">

Switch to executing as a public user and all comment author avatars become:
"http://0.gravatar.com/avatar/?s=96&d=identicon&r=g" which is not correct.
<img width="1007" alt="Screenshot 2022-06-25 at 09 45 50" src="https://user-images.githubusercontent.com/4234033/175765840-8f7625d4-b75b-4dae-9675-78b9caf0adf0.png">

I also observe the same behaviour when querying comments directly:
<img width="987" alt="Screenshot 2022-06-25 at 10 14 38" src="https://user-images.githubusercontent.com/4234033/175766821-c9c9980f-44c3-42ad-aec6-9932a18b8bf1.png">

Hope this helps, please let me know if you need more info on this.
Thank you!

**Comment by romcik** [comment]
Hello, same issue here. Avatar URL differs when query runs as anonymous or not. Any progress here?

![image](https://user-images.githubusercontent.com/15350220/184632061-d1dbbcc1-39d1-43b9-b03b-44a91b664106.png)
![image](https://user-images.githubusercontent.com/15350220/184632148-e4b53c4f-9bfa-4d3c-8194-76ce4f81bc54.png)

**Comment by romcik** [comment]
problem is here:
\wp-graphql\src\Type\ObjectType\CommentAuthor.php on line 80
`$avatar = get_avatar_data( $comment_author->email, $avatar_args );
`
avatar needs comment author email to generate URL, but CommentAuthor model provides email only when logged user has capability "moderate_comments".
\wp-graphql\src\Model\CommentAuthor.php line 61
```
'email'      => function () {
					return current_user_can( 'moderate_comments' ) && ! empty( $this->data->comment_author_email ) ? $this->data->comment_author_email : null;
				},
```

**Comment by romcik** [comment]
fast solution without change of plugin code, but exposes commenter's email to public.
`add_filter('graphql_pre_return_field_from_model', function ($result, $key, $model_name, $data = null, $visibility = null, $owner = null, $current_user = null) {
    if ($model_name == "CommentAuthorObject" && $key == "email") {
        return $data->comment_author_email;
    }
    return $result;
}, 10, 7);`

or add 'email_private' field to CommentAuthor for internal use and then get avatar this way:
`$avatar = get_avatar_data( $comment_author->email_private, $avatar_args );`
\wp-graphql\src\Model\CommentAuthor.php line 61

**Comment by romcik** [comment]
and one more strange behavior. Same query returns type CommentAuthor or User depending if current user is logged.
For anonymous is author type of CommentAuthor. BTW also databaseId of CommentAuthor is same as commentId. 
![image](https://user-images.githubusercontent.com/15350220/184645314-7bf9d955-2e58-424f-82f8-14a67032b8f2.png)

for logged is author type of User 
![image](https://user-images.githubusercontent.com/15350220/184645403-c6699371-ab03-4248-add4-b645c11da5be.png)

**Comment by justlevine** [comment]
Hey @romcik - great sleuthing!

> and one more strange behavior. Same query returns type CommentAuthor or User depending if current user is logged.
For anonymous is author type of CommentAuthor. BTW also databaseId of CommentAuthor is same as commentId.

This [should have been fixed](https://github.com/wp-graphql/wp-graphql/pull/2452) in v1.9.0. What version of WPGraphQL did you run your tests on?

**Comment by romcik** [comment]
ou, sorry, ok. testing  1.8.7

**Comment by justlevine** [comment]
It seems that by fixing the returned model in 1.9.0, the avatars  for `Users` are fixed when unauthenticated, but `CommentAuthors` still arent.

![image](https://user-images.githubusercontent.com/29322304/184647876-e9dd237e-5552-46ff-840b-9882ff4fe973.png)

**Comment by romcik** [comment]
I confirm too. I used temporary workaround. 
`$avatar = get_avatar_data( $comment_author->email_private, $avatar_args ); // in CommentAuthor.php line 80
`

and 
```
add_filter('graphql_model_prepare_fields', function ($fields, $model_name, $data = null, $visibility = null, $owner = null, $current_user = null) {
    $fields['email_private'] = function () use ($data) { 
        return $data->comment_author_email;
    };
    return $fields;
}, 10, 6);
```

**Comment by justlevine** [comment]
@romcik , @umutesen #2472 should do the trick. If you can, please test and confirm 😎

**Comment by jasonbahl** [comment]
Ah! Good catch here. Thanks for the PR @justlevine

**Comment by romcik** [comment]
> @romcik , @umutesen #2472 should do the trick. If you can, please test and confirm 😎
it works. Nice patch :-)

**Comment by umutesen** [comment]
@justlevine I can confirm it is working. Thank you!

**Comment by justlevine** [comment]
Fixed in #2472
